src/App.js

```diff
@@ -7,18 +7,33 @@ import exercises from './db';
 export default function App() {
   const [filter, setFilter] = useState('Alle');
 
+  const [exercise, setExercise] = useState(exercises);
+
   function handleFilter(stringToFilter) {
     setFilter(stringToFilter);
   }
 
+  function changeBookmark(id, bookmarked) {
+    exercises.forEach(() => {
+      setExercise(current =>
+        current.map(obj => {
+          if (obj.id === id) {
+            return {...obj, bookmarked: bookmarked};
+          }
+          return obj;
+        })
+      );
+    });
+  }
```

Comment:
You don't need two loops at this point. You can solve your problem with just using the map methode.

Reply:
Better naming would be handleBookmark instead of changeBookmark.